### DISCUSSION TITLE
Groovy graphqlIntrospectSchema failing on kotlin.jvm.internal.MutablePropertyReference

### DISCUSSION THREAD

**Discussion by FidasL** [ExpediaGroup/graphql-kotlin#1233] [General] [answered]
**Library Version**
4.1.1

Hello. Our gradle plugin is failing on task graphqlIntrospectSchema. Getting this error:
```
Execution failed for task ':graphqlIntrospectSchema'.
> There was a failure while executing work items
   > A failure occurred while executing com.expediagroup.graphql.plugin.gradle.actions.IntrospectSchemaAction
      > kotlin.jvm.internal.MutablePropertyReference1Impl.<init>(Ljava/lang/Class;Ljava/lang/String;Ljava/lang/String;I)V
```

`build.gradle` configuration:

```
plugins {
    id("com.expediagroup.graphql") version "4.1.1"
}

graphql {
    // generates schema.graphql to buildDir/schema.graphql
    graphqlIntrospectSchema {
        endpoint = "http://localhost:8080/graphql"
    }

    graphqlGenerateClient {
        schemaFileName = "$projectDir/resources/schema.graphql"
        packageName = "com.example.client.graphql.generated"
        queryFileDirectory = "$projectDir/resources/graphql/"
        allowDeprecatedFields = false
    }
}
```

Do we need to provide some extra parameters when introspecting schema, or this is your issue?

**Comment by dariuszkuc** [comment, marked as answer]
Hello 👋 
I just tried the above configuration and with few minimal tweaks it worked for me. I could not reproduce your error.

```build.gradle
plugins {
    id("org.jetbrains.kotlin.jvm") version '1.4.32'
    id("com.expediagroup.graphql") version "4.1.1"
}

group 'org.example'
version '1.0-SNAPSHOT'

repositories {
    mavenCentral()
}

dependencies {
    implementation "org.jetbrains.kotlin:kotlin-stdlib"
    implementation "com.expediagroup:graphql-kotlin-spring-client:4.1.1" // spring or ktor client is required for compilation
}

graphql {
    graphqlIntrospectSchema {
        endpoint = "http://localhost:8080/graphql"
    }

    graphqlGenerateClient {
        schemaFile = graphqlIntrospectSchema.outputFile // changed to use the output from other task
        packageName = "com.example.client.graphql.generated"
        queryFileDirectory = "$projectDir/resources/graphql/"
        allowDeprecatedFields = false
        dependsOn("graphqlIntrospectSchema") // added explicit dependency on task so they run in order
    }
}
```

That being said - `graphql` extension provides a shorthand for configuring those two tasks so you could simplify your config (see: https://opensource.expediagroup.com/graphql-kotlin/docs/plugins/gradle-plugin-usage#minimal-client-configuration-example). Refer to our [documentation](https://opensource.expediagroup.com/graphql-kotlin/docs/plugins/gradle-plugin-usage) for additional details.

**Reply by FidasL** [reply]
Hello. Thank you for your reply!

I got same error with your simple example. Do you think, it could be caused by Gradle version? I am using Gradle 6.7

**Reply by dariuszkuc** [reply]
Above should work with Groovy scripts (`build.gradle`) it won't work if you use Kotlin scripts (`build.gradle.kts`). See the docs for configuration examples.

You could also move task definitions outside of `graphql` extension as that might be causing your issue (or simply configure extension instead of tasks -> again see the docs).
